large-message-core/src/test/java/com/bakdata/kafka/AmazonS3LargeMessageClientRoundtripTest.java

```diff
@@ -0,0 +1,100 @@
+package com.bakdata.kafka;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.apache.kafka.common.header.Header;
+import org.apache.kafka.common.header.internals.RecordHeaders;
+import org.apache.kafka.common.serialization.Serdes;
+import org.apache.kafka.common.serialization.Serializer;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.google.common.collect.ImmutableMap;
+
+import io.confluent.common.config.ConfigDef;
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.CreateBucketRequest;
+
+class AmazonS3LargeMessageClientRoundtripTest extends AmazonS3IntegrationTest {
+
+    private static final String TOPIC = "output";
+    private static final Serializer<String> STRING_SERIALIZER = Serdes.String().serializer();
+
+    private static byte[] serialize(final String s) {
+        return STRING_SERIALIZER.serialize(null, s);
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideParameters")
+    void shouldRoundtrip(final boolean isKey, final String compressionType) {
+        final String bucket = "bucket";
+        final String basePath = "s3://" + bucket + "/base/";
+        final Map<String, Object> properties = ImmutableMap.<String, Object>builder()
+                .put(AbstractLargeMessageConfig.MAX_BYTE_SIZE_CONFIG, 0)
+                .put(AbstractLargeMessageConfig.BASE_PATH_CONFIG, basePath)
+                .put(AbstractLargeMessageConfig.COMPRESSION_TYPE_CONFIG, compressionType)
+                .build();
+        final S3Client s3 = this.getS3Client();
+        s3.createBucket(CreateBucketRequest.builder().bucket(bucket).build());
+        final LargeMessageStoringClient storer = this.createStorer(properties);
+
+        final LargeMessageRetrievingClient retriever = this.createRetriever();
+
+        RecordHeaders headers = new RecordHeaders();
```

Comment:
Variable type can be weakened to Headers. Also final (apply code style/clean up)